src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -272,10 +276,49 @@ impl BDA {
     where
         F: Read + Seek + SyncAll,
     {
-        StaticHeader::setup(f).map(|sh| sh.map(|sh| (sh.pool_uuid, sh.dev_uuid)))
+        match StaticHeader::setup(f) {
+            Ok(Some(SetupResult::Ok(sh))) => Ok(Some((sh.pool_uuid, sh.dev_uuid))),
+            Ok(Some(SetupResult::OkWithError(sh, err))) => {
+                setup_warn(&sh, err);
+                Ok(Some((sh.pool_uuid, sh.dev_uuid)))
+            }
+            Ok(None) => Ok(None),
+            Err(err) => Err(err),
+        }
     }
 }
 
+fn setup_warn(header: &StaticHeader, err: StratisError) {
```

Comment:
Please add some explanatory header comments.

Reply:
I added a comment. Considering the function was just emitting a warning, I didn't think it was necessary to kinda repeat what it was doing but I guess a comment cannot hurt. Also, don't hesitate if the comment isn't well written.